recipes-core/ncurses/ncurses_%.bbappend

```diff
@@ -0,0 +1 @@
+CACHED_CONFIGUREVARS_append_toolchain-clang = " cf_cv_prog_CC_c_o=yes cf_cv_prog_CXX_c_o=yes"
```

Comment:
I think I narrowed it down further. My host is running CentOS 7 64 bit, I'm testing with MACHINE=wandboard, including oe-core, meta-oe, meta-clang, meta-fsl-arm, and meta-fsl-arm-extra. I *do not* have glibc-devel.i686 installed. With that setup I see in the build log:

configure:2524: checking whether arm-poky-linux-gnueabi-clang understands -c and -o together
configure:2540: arm-poky-linux-gnueabi-clang -c conftest.c -o conftest2.o >&5
In file included from conftest.c:1:
In file included from /usr/include/stdio.h:27:
In file included from /usr/include/features.h:399:
/usr/include/gnu/stubs.h:7:11: fatal error: 'gnu/stubs-32.h' file not found
# include <gnu/stubs-32.h>
          ^
1 error generated.
configure:2543: $? = 1
configure:2562: result: no

As you can see, it's checking the *host* for gnu/stubs-32.h. It's not there on my host because I don't have glibc-devel.i686 installed. If I install glibc-devel.i686 the build succeeds without this change. I'll update the commit message with my new discoveries.